lib/ansible/plugins/action/template.py

```diff
@@ -135,8 +135,8 @@ def run(self, tmp=None, task_vars=None):
             new_task.args.pop('variable_end_string', None)
             new_task.args.pop('trim_blocks', None)
             try:
-                tempdir = tempfile.mkdtemp(dir=C.DEFAULT_LOCAL_TMP)
-                result_file = os.path.join(tempdir, os.path.basename(source))
+                local_tempdir = tempfile.mkdtemp(dir=C.DEFAULT_LOCAL_TMP)
```

Comment:
Isn't the problem here that the try-block is masking the error that tempfile.mkdtemp may be throwing ?

Reply:
Yeah.  So  I contemplated two different fixes.  (1) Move the call to mkdtemp() outside of the try: finally.  or (2) test in the finally whether tempdir is set.  I think I went with this method because there's similar code else where but now that you mention it, I think it's anti-patterm.  Moving it outside of the try: block is the better solution.